aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLOperation.java

```diff
@@ -142,6 +142,9 @@ public void onResponse(@NonNull Call call, @NonNull Response response) {
                     return;
                 }
             }
+            if(response.code() >= 400) {
```

Comment:
5xx level errors should be retried.  

```suggestion
            if(response.code() >= 400 && response.code() <= 499) {
```